server/src/arcs-master-app.ts

```diff
@@ -40,18 +42,32 @@ class ArcsMasterApp extends AppBase {
 
     try {
       const cloud = CloudManager.forGCP();
-      const container: Container | null = await cloud.containers().find(fingerprint);
-      if (container != null) {
-        const url = container.url();
-        if (url === 'pending') {
-          res.send('Creation still in progress, refresh for status.');
+      const disk = await cloud.disks().find(fingerprint);
+      if (disk) {
+        const attached = await disk.isAttached();
+        const wrappedKey  = await disk.wrappedKeyFor(fingerprint);
+
+        if (attached) {
+          const container: Container | null = await cloud.containers().find(fingerprint);
+          if (container != null) {
+            const url = container.url();
+            if (url === 'pending') {
+              res.send('{"id": "' + fingerprint + '", "status": "pending"}');
+            } else {
+              res.send('{"id": "' + fingerprint + '", "status": "attached", "url" : "' + url + '"}');
+            }
+            return;
+          }
+          res.send('{"id": "' + fingerprint + '", "status": "container not found"}');
```

Comment:
Two nits: (1) these might be easier to read (less deep nesting) as early-outs (here and below), and (2) you can use template literals to make these more readable, like wrap in backticks and use `${fingerprint}` instead of string concat `+`, same rest of PR (neither are a big deal, can do in future)